lib/std/builtin.zig

```diff
@@ -1000,6 +1000,17 @@ pub const ExternOptions = struct {
     is_thread_local: bool = false,
     is_dll_import: bool = false,
     relocation: Relocation = .any,
+    decoration: ?Decoration = null,
```

Comment:
Still not sure about the best interface here. Should this just be
```zig
    location: ?u32 = null,
    binding: ?u32 = null,
    set: ?u32 = null,
```
to correspond more directly to the SPIR-V? But that provides no type safety around the fact that (I think) locations and descriptors are mutually exclusive, and bindings/sets must always be specified together.

Another complexity here is that SPIR-V does not actually require a name for the binding. So, should the `name` field of this struct be made optional? But if you don't set a name, you _must_ set a location or descriptor, so should the whole thing just be a tagged union of `{ name, location, descriptor }`? But that's not backwards compatible and makes all other users of `@extern` think about SPIR-V.

Reply:
Perhaps, i suggest to keep it like this and bring this topic in the zig repo PR.